src/test/java/org/assertj/core/internal/strings/Strings_assertContainsExactlyInAnyOrder_Test.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2017 the original author or authors.
+ */
+
+package org.assertj.core.internal.strings;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.StandardComparisonStrategy;
+import org.assertj.core.internal.Strings;
+import org.assertj.core.internal.StringsBaseTest;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.assertj.core.error.ShouldContainExactlyInAnyOrder.*;
+import static org.assertj.core.error.ShouldHaveSameSizeAs.*;
+import static org.assertj.core.internal.ErrorMessages.*;
+import static org.assertj.core.test.TestData.*;
+import static org.assertj.core.util.FailureMessages.*;
+
+/**
+ * Tests for <code>{@link Strings#assertContainsExactlyInAnyOrder(AssertionInfo, CharSequence, CharSequence[])}</code>.
+ *
+ * @author Billy Yuan
+ */
+public class Strings_assertContainsExactlyInAnyOrder_Test extends StringsBaseTest {
+  String actual = "Practice makes perfect!";
+
```

Comment:
add some tests where actual and expected have duplicates (we had some bugs in iterable assertions related to duplicates)